### ISSUE TITLE
Add error handling coverage for Sass mixins & functions

### ISSUE THREAD

**Issue by elseloop** [isitedesign/Phoenix#72]
It could be good to take advantage of Sass's built-in error handling and write in some coverage for Phoenix's Sass mixins and functions. While the Gulp tasks already give some pretty clear messaging, we can make it a lot more specific, which might be good for people coming to Phoenix for the first time.

For instance, if you were to pass a string to the `parseInt()` function, Gulp will tell you "error: invalid operands for multiplication." It's clear if you have context for it, but it could be confusing on first glance. If we add something at the head of the function to test if the passed parameter is actually a number, we can tell someone as much if what they pass isn't. Something like:

``` scss
@function parseInt($n) {
    @if type_of($n) != "number" {
        @warn "The parseInt() function can only be passed numbers. You passed in #{$n}."
    }
    @return $n / ($n * 0 + 1);
}
```

As far as I can tell, Lib-Sass doesn't support Sass's `@error` directive yet (came in at Sass 3.4), but it does support `@warn`, which might actually have a benefit over `@error` for this use case, I think, in that it will output our message ahead of the Gulp/Plumber error message and not stop that from being printed to the console (which I think `@error` would, as it instantly throws a fatal error).

Anyway, I'm happy to start going through and writing these in if people think it'd be useful to have.

**Comment by ajmueller** [comment]
This looks like another ticket that is dependent on #105.

**Comment by elseloop** [comment]
As of [1.0.2](https://github.com/isitedesign/Phoenix/releases/tag/1.0.2), while we’ve gained Sass support for `@error`, it looks like the build system updates and enhancements made by @kamsar in #97 have eliminated our ability to output Sass errors and warnings. In testing this, it seems Plumber throws a fatal error before the Sass `@error` message an be parsed, so the custom messages in our `.scss` files don't hit the console before `gulp-sass` outputs its message and kills the process. Basically, the Gulp pipeline is so fast as to not need our middling Sass errors.

I'm going to put this on hold for now, since the `gulp-sass` error messaging is clear enough to be helpful and we're therefore not missing any features with this. I am, however, also leaving it open, as the idea of writing error coverage into our mixins and functions is still appealing.

**Comment by ajmueller** [comment]
As we noticed in #114, `@warn` is outputting twice.  As part of adding better warnings to our functions and mixins, could you investigate the double output further?
